lib/src/repo.rs

```diff
@@ -965,3 +978,14 @@ impl MutableRepo {
         );
     }
 }
+
+/// Error from attempts to check out the root commit for editing
+#[derive(Debug, Copy, Clone, Error)]
+#[error("Opened root commit")]
+pub struct OpenedRootCommit;
+
+impl From<OpenedRootCommit> for BackendError {
+    fn from(err: OpenedRootCommit) -> Self {
+        BackendError::Other(err.to_string())
+    }
+}
```

Comment:
This feels backwards - `BackendError` is a lower-level error. Is this needed only for the change in `rewrite.rs`? After reviewing the code for a few minutes, I think it should be correct to `unwrap()` there.

Longer term, we should return a more specific error there instead of the low-level `BackendError`.

Reply:
Sounds good to me. Even if the unwrap turns out to be wrong, it's certainly not a step backwards in correctness. It did seem a bit odd to have something apparently backend-independent returning `BackendError`.